app/gui/src/dashboard/layouts/DriveBar.tsx

```diff
@@ -128,53 +118,64 @@ export default function DriveBar(props: DriveBarProps) {
       pasteData
     : null
 
+  const getTargetDirectory = useEventCallback(() => driveStore.getState().targetDirectory)
+  const rootDirectoryId = useRootDirectoryId(backend, category)
+
+  const newFolderRaw = useNewFolder(backend, category)
+  const newFolder = useEventCallback(async () => {
+    const parent = getTargetDirectory()
+    return await newFolderRaw(parent?.directoryId ?? rootDirectoryId, parent?.path)
+  })
+  const uploadFilesRaw = useUploadFiles(backend, category)
+  const uploadFiles = useEventCallback(async (files: readonly File[]) => {
+    const parent = getTargetDirectory()
+    await uploadFilesRaw(files, parent?.directoryId ?? rootDirectoryId, parent?.path)
+  })
+  const newSecretRaw = useNewSecret(backend, category)
+  const newSecret = useEventCallback(async (name: string, value: string) => {
+    const parent = getTargetDirectory()
+    return await newSecretRaw(name, value, parent?.directoryId ?? rootDirectoryId, parent?.path)
+  })
+  const newDatalinkRaw = useNewDatalink(backend, category)
+  const newDatalink = useEventCallback(async (name: string, value: unknown) => {
+    const parent = getTargetDirectory()
+    return await newDatalinkRaw(name, value, parent?.directoryId ?? rootDirectoryId, parent?.path)
+  })
+  const newProjectRaw = useNewProject(backend, category)
+  const newProjectMutation = useMutation({
+    mutationKey: ['newProject'],
+    mutationFn: async ([templateId, templateName]: [
+      templateId: string | null | undefined,
+      templateName: string | null | undefined,
+    ]) => {
+      const parent = getTargetDirectory()
+      return await newProjectRaw(
+        { templateName, templateId },
+        parent?.directoryId ?? rootDirectoryId,
+        parent?.path,
+      )
+    },
+  })
```

Comment:
@MrFlashAccount currently when we want to use the new hooks as a mutation they are wrapped into a mutation manually